pkg/operator/starter.go

```diff
@@ -0,0 +1,95 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cert-manager-operator/pkg/controller/deployment"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	TargetNamespace   = "openshift-cert-manager"
+	OperatorNamespace = "openshift-cert-manager-operator"
+)
+
+func RunOperator(ctx context.Context, cc *controllercmd.ControllerContext) error {
+	kubeClient, err := kubernetes.NewForConfig(cc.ProtoKubeConfig)
+	if err != nil {
+		return err
+	}
+
+	configClient, err := configv1client.NewForConfig(cc.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	genericOperatorClient, dynamicInformers, err := genericoperatorclient.NewClusterScopedOperatorClient(
+		cc.KubeConfig, operatorv1.GroupVersion.WithResource("certmanagers"))
+	if err != nil {
+		return err
+	}
```

Comment:
maybe we can have our own operator client created later but this is fine for now

Reply:
yes agreed

Reply:
after thought: let's leave this out for now as the unstuctured.Unstructured informer will otherwise complain with `failed to list *unstructured.Unstructured: the server could not find the requested resource`.